lib/util/db.js

```diff
@@ -239,7 +241,33 @@ const queryFuncs = (db, obj) => {
     for (let i = 0; i < xs.length; i += 1) result[i] = f(xs[i]);
     return result;
   };
-  obj.stream = (s) => new Promise((resolve, reject1) => db.stream(s, resolve).catch(reject1));
+
+  obj.stream = (s) => new Promise((resolve, reject1) => db.stream(s, stream => {
+    try {
+      const twoMinutes = 2 * 60 * 1000;
+      const streamTimeout = setTimeout(() => {
+        stream.destroy(new Error('Stream timed out.'));
+      }, twoMinutes);
+
+      const wrappedStream = stream.pipe(new Transform({
+        objectMode: true,
+        transform(data, encoding, cb) {
+          streamTimeout.refresh();
+          this.push(data);
+          cb();
+        },
+      }));
+
+      stream.on('close', () => {
+        clearTimeout(streamTimeout);
+      });
+
+      resolve(wrappedStream);
+    } catch (error) {
+      reject1(error);
+    }
+  }).catch(reject1));
```

Comment:
We're going to include this change in the patch release (as part of #604) because that ended up seeming like the best way to address the database connection leaks for now.